### PR TITLE
DO NOT MERGE ci: Add branching workflow scripts

### PR DESCRIPTION
**Pull request by mstruebing**
## Jira Ticket
Before creating this pull request, make sure you have a separate ticket for the docs team to track their work. If you need assistance, ask in the #documentation Slack channel.

## Description of changes being made

This is a model how our ci implementation could look like. There are some TODOS open but I wanted to push it already so that @vacas5 could have a look and make suggestions.

I already created three branches (fresh out of main: `production`, `beta).

## Checklist
- [ ] Change all affected versions, if applicable (e.g. 1.13, 2.0, 2.1).
- [ ] Test all commands and procedures, if applicable.
- [ ] Create your PR against `main`. 
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](https://wiki.d2iq.com/display/ENG/Contributing+to+Docs) for more information.

### CODE REVIEW

**Review by cmastr — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -42,12 +47,29 @@ pipeline {
       }
     }
 
+    stage("Create Pull Requests") {
```
Comment (vacas5) [inline Jenkinsfile:50]: can we handle the automation piece in a separate PR? I want to discuss if its even necessary a bit further. I think I'd like to have @cmastr try making one this first go round so she can understand how easy it is to do within Github.

Creating it when it is time to release is going to end up with less open PRs that need to be rebased, etc.

File: Jenkinsfile [hunk 2]
```diff
@@ -42,12 +47,29 @@ pipeline {
...
+    }
+
     stage("Build & Deploy Docs") {
       environment {
-        ALGOLIA_UPDATE = "${main ? 'true' : ''}"
+        ALGOLIA_UPDATE = "${isProduction ? 'true' : ''}"
         AWS_DEFAULT_REGION = "us-west-2"
         BUCKET = "docs-d2iq-com-${bucket}"
-        PRINCIPAL = "arn:aws:iam::139475575661:role/Jenkins/Jenkins-S3-DOCS-${main ? 'Production' : 'Development'}"
+        // TODO: What does that mean?
```
Comment (vacas5) [inline Jenkinsfile:71]: I agree its worth checking with devprod to find out what this is about

File: Jenkinsfile [hunk 3]
```diff
@@ -1,15 +1,20 @@
 #!/usr/bin/env groovy
 
-boolean main = env.BRANCH_NAME == "main"
-boolean beta = env.BRANCH_NAME == "private-beta"
-def bucket   = main ? "production"
-             : beta ? "staging"
+boolean isProduction = env.BRANCH_NAME == "production"
+boolean isBeta = env.BRANCH_NAME == "beta"
+boolean isDevelop = env.BRANCH_NAME = "develop"
```
Comment (vacas5) [inline Jenkinsfile:5]: does jenkins provide these automatically? if so, that's great

**Review by vacas5 — CHANGES_REQUESTED** [review]
I think you wanted to delete the `develop` branch and create some other ones, yes?

File: Jenkinsfile (hunk 3 above)
Comment (mstruebing) [reply Jenkinsfile:5]: I assume, I didn't changed anything in the Jenkins UI already, will need to check.

**Comment by mstruebing** [comment]
> I think you wanted to delete the develop branch and create some other ones, yes?

There was no develop branch present?
I created one, and this acts as our base/main branch for everything I targeted this PR on the develop branch.

with `git diff develop main` you can see that it is no difference (at least right now).
I created a beta and production branch too which doesn't existed before.

File: Jenkinsfile (hunk 1 above)
Comment (mstruebing) [reply Jenkinsfile:50]: Removed

File: Jenkinsfile (hunk 3 above)
Comment (mstruebing) [reply Jenkinsfile]: It did before, why should it change?

File: Jenkinsfile [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
-boolean beta = env.BRANCH_NAME == "private-beta"
-def bucket   = main ? "production"
-             : beta ? "staging"
+boolean isProduction = env.BRANCH_NAME == "production"
+boolean isBeta = env.BRANCH_NAME == "beta"
+boolean isMain = env.BRANCH_NAME = "main"
+
+def bucket   = isProduction ? "production"
+             : isBeta ? "beta"
+             : isMain ? "main"
```
[the remark this answers is not in the document]
Comment (mstruebing) [reply Jenkinsfile]: Asked in #devprod. https://mesosphere.slack.com/archives/C5U03P5T6/p1630483711012800

**Comment by mstruebing** [comment]
Right before merging this we need to adjust the jenkins multi branch configuration to use the newly created branches (https://jenkins.mesosphere.com/service/jenkins/job/dcos-docs-site/configure)

I asked in #devprod about domain, bucket and principal.
I think that everything we can do right now is to wait for answers from #devprod and may adjust things if needed.

**Comment by mstruebing** [comment]
Askes in #sre/#devprod about all our open questions.

**Pull request merged**